### ISSUE TITLE
gaCookies using potentially fragile mechanism

### ISSUE THREAD

**Issue by shermozle** [snowplow/snowplow-javascript-tracker#578]
[gaCookies](https://github.com/snowplow/snowplow-javascript-tracker/blob/0ba1bad19e276d381ffe214669042dde17a4d9fd/src/js/tracker.js#L1202) just takes the entire Google Analytics cookie out and plugs it into a context.

GA [provides a method](https://developers.google.com/analytics/devguides/collection/analyticsjs/cookies-user-id#getting_the_client_id_from_the_cookie) and callback to do this properly. We should probably use this. Possibly keep both for backwards compatibility? This probably makes the modelling stage a bit easier too since there's more than just the client ID in there.

**Comment by alexanderdean** [comment]
Changing this would introduce a dependency on the Google Analytics tracker though - is that what we want? What if somebody has switched off GA but still wants to track the cookie...

**Comment by shermozle** [comment]
Good point. Perhaps a mechanism to push a custom value in and point to that GA article in docs?

The point Google make is that you can't rely on the cookie being there. Perhaps gaCookies could be generalised to a "grab all this list of cookies" so it could pull in other first-party cookies too?
